tests/test_dcrAssembleCoadd.py

```diff
@@ -0,0 +1,295 @@
+        elevation = 65.*afwGeom.degrees
+        visitInfo = self.makeDummyVisitInfo(azimuth, elevation)
+        wcs = self.makeDummyWcs(rotAngle, visitInfo=visitInfo)
+        dcrShift = DcrAssembleCoaddTask.dcrShiftCalculate(visitInfo, wcs,
+                                                          self.config.lambdaEff,
+                                                          self.config.filterWidth,
+                                                          self.config.dcrNSubbands)
+        refShift = [dcr(dx=-0.55832265311722795, dy=-0.32306512577396451),
+                    dcr(dx=-0.018151534656568987, dy=-0.010503116422151829),
+                    dcr(dx=0.36985291822812622, dy=0.21400990785188412)]
+        for shiftOld, shiftNew in zip(refShift, dcrShift):
+            self.assertFloatsAlmostEqual(shiftOld.dx, shiftNew.dx, rtol=1e-6, atol=1e-8)
+            self.assertFloatsAlmostEqual(shiftOld.dy, shiftNew.dy, rtol=1e-6, atol=1e-8)
+
+    def testRotationAngle(self):
+        """! Test that he sky rotation angle is consistently computed.
+
+        The rotation is compared to pre-computed values.
+        """
+        cdRotAngle = Angle(0.)
+        azimuth = 130.*afwGeom.degrees
+        elevation = 70.*afwGeom.degrees
+        visitInfo = self.makeDummyVisitInfo(azimuth, elevation)
+        wcs = self.makeDummyWcs(cdRotAngle, visitInfo=visitInfo)
+        rotAngle = DcrAssembleCoaddTask.calculateRotationAngle(visitInfo, wcs)
+        refAngle = Angle(-0.9344289857053072)
+        self.assertAnglesNearlyEqual(refAngle, rotAngle)
+
+    def testConditionDcrModelNoChange(self):
+        """! Conditioning should not change the model if it is identical to the reference.
+
+        This additionally tests that the variance and mask planes do not change.
+        """
+        refModels = [model.clone() for model in self.dcrModels]
+        DcrAssembleCoaddTask.conditionDcrModel(refModels, self.dcrModels, self.bbox, gain=1.)
+        for model, refModel in zip(self.dcrModels, refModels):
+            self.assertMaskedImagesEqual(model, refModel)
+
+    def testConditionDcrModelWithChange(self):
+        """! Verify the effect of conditioning when the model changes by a known amount.
+
+        This additionally tests that the variance and mask planes do not change.
+        """
+        refModels = [model.clone() for model in self.dcrModels]
+        for model in self.dcrModels:
+            model.getImage().getArray()[:, :] *= 3.
+        DcrAssembleCoaddTask.conditionDcrModel(refModels, self.dcrModels, self.bbox, gain=1.)
+        for model, refModel in zip(self.dcrModels, refModels):
+            refModel.getImage().getArray()[:, :] *= 2.
+            self.assertMaskedImagesEqual(model, refModel)
+
+    def testShiftImagePlane(self):
+        """! Verify the shift calculation for the image and variance planes.
+
+        The shift of the mask plane is tested separately since it is calculated differently.
```

Comment:
But the set up for the mask shift test is the same. You need to factor out the common code one way or another.